op-deployer/pkg/deployer/inspect/semvers.go

```diff
@@ -0,0 +1,198 @@
+package inspect
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"math/big"
+	"regexp"
+	"time"
+
+	"github.com/ethereum-optimism/optimism/op-chain-ops/script"
+
+	"github.com/ethereum-optimism/optimism/op-deployer/pkg/deployer/broadcaster"
+	"github.com/ethereum-optimism/optimism/op-deployer/pkg/deployer/pipeline"
+	"github.com/ethereum-optimism/optimism/op-service/ioutil"
+	"github.com/ethereum-optimism/optimism/op-service/jsonutil"
+	oplog "github.com/ethereum-optimism/optimism/op-service/log"
+	"github.com/ethereum-optimism/optimism/op-service/predeploys"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/holiman/uint256"
+	"github.com/urfave/cli/v2"
+)
+
+var versionSelector = []byte{0x54, 0xfd, 0x4d, 0x50}
+
+func L2SemversCLI(cliCtx *cli.Context) error {
+	cliCfg, err := readConfig(cliCtx)
+	if err != nil {
+		return err
+	}
+
+	ctx, cancel := context.WithTimeout(cliCtx.Context, time.Minute)
+	defer cancel()
+
+	logCfg := oplog.ReadCLIConfig(cliCtx)
+	l := oplog.NewLogger(oplog.AppOut(cliCtx), logCfg)
+	oplog.SetGlobalLogHandler(l.Handler())
+
+	globalState, err := pipeline.ReadState(cliCfg.Workdir)
+	if err != nil {
+		return fmt.Errorf("failed to read intent: %w", err)
+	}
+	chainState, err := globalState.Chain(cliCfg.ChainID)
+	if err != nil {
+		return fmt.Errorf("failed to find chain state: %w", err)
+	}
+
+	intent := globalState.AppliedIntent
+	if intent == nil {
+		return fmt.Errorf("can only run this command following a full apply")
+	}
+	if chainState.Allocs == nil {
+		return fmt.Errorf("chain state does not have allocs")
+	}
+
+	artifactsFS, cleanup, err := pipeline.DownloadArtifacts(ctx, intent.L2ContractsLocator, pipeline.LogProgressor(l))
+	if err != nil {
+		return fmt.Errorf("failed to download L2 artifacts: %w", err)
+	}
+	defer func() {
+		if err := cleanup(); err != nil {
+			l.Warn("failed to clean up L2 artifacts", "err", err)
+		}
+	}()
+
+	host, err := pipeline.DefaultScriptHost(
+		broadcaster.NoopBroadcaster(),
+		l,
+		common.Address{19: 0x01},
+		artifactsFS,
+		0,
+	)
+	if err != nil {
+		return fmt.Errorf("failed to create script host: %w", err)
+	}
+	host.ImportState(chainState.Allocs.Data)
+
+	addr := common.Address{19: 0x01}
+
+	type contractToCheck struct {
+		Address common.Address
+		Name    string
+	}
+
+	contractsOutput := make(map[string]string)
+
+	// The gov token and the proxy admin do not have semvers.
+	contracts := []contractToCheck{
```

Comment:
Looks like we are also not checking the ERC20 and ERC721 tokens?

Reply:
you mean the mintable ones? those are checked below

Reply:
nevermind yea I see them below, separate from the loop